File: angular-frontend/src/app/services/shopping/payment-method/payment-method.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {PaymentMethod} from "../../../common/shopping/payment-method";

@Injectable({
  providedIn: 'root'
})
export class PaymentMethodService {

  private baseUrl = 'http://localhost:6969/api/';

  constructor(private httpClient: HttpClient) { }

  getPaymentMethodById(id: number) {
    return this.httpClient.get<PaymentMethod>(this.baseUrl + 'admin/getPaymentMethodById?paymentMethodId=' + id)
  }

  getAllPaymentMethods() {
    return this.httpClient.get<PaymentMethod[]>(this.baseUrl + 'admin/getAllPaymentMethods');
  }

  savePaymentMethod(paymentMethod: PaymentMethod, imageFile: File | null) {
    let formData = new FormData();
    formData.append('paymentMethod', new Blob([JSON.stringify(paymentMethod)], {type: 'application/json' }) );
    if (imageFile) {
      formData.append('image', imageFile);
    }
    return this.httpClient.post(this.baseUrl + 'admin/savePaymentMethod', formData);
  }
}

File: spring-boot-backend/src/main/java/com/eafc/springbootbackend/services/shopping/PaymentMethodService.java
package com.eafc.springbootbackend.services.shopping;

import com.eafc.springbootbackend.entities.shopping.PaymentMethod;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.Collection;

public interface PaymentMethodService {

    PaymentMethod getPaymentMethodById(Long paymentMethodId);

    Collection<PaymentMethod> getAllPaymentMethods();
    void savePaymentMethod(PaymentMethod paymentMethod, MultipartFile image) throws IOException;
}
